consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -157,6 +180,6 @@ public ConsulUpdatingListenerBuilder consulToken(String consulToken) {
      */
     public ConsulUpdatingListener build() {
         return new ConsulUpdatingListener(consulClientBuilder.build(), serviceName, serviceEndpoint,
-                                          checkUri, checkMethod, checkInterval);
+                                          checkUri, checkMethod, checkInterval, tagsBuilder.build().asList());
```

Comment:
Can we just use `Set` for the tags?

Reply:
It is possible if we change the type of `Service.tags` to `List<String>`.

Reply:
In theory, the Consul API allows for duplicate tags, since it is passed as a JSON list of strings. In practice though, it's unlikely that this is the user's desired behaviour (see issue below).

https://github.com/hashicorp/consul/issues/2329

I kept `Service.tags` as a `List<String>` to match the Consul API, in case a service returned by Consul does in fact contains duplicate tags (although Armeria does not currently contains any read call to the service API). However, I kept it a `Set<String>` here, since as I mentioned, I don't expect users to voluntarily want to add the same tags multiple times.

So I guess there are 3 options:

1. Follow the Consul API strictly and make everything a `List<String>`, potentially letting users add the same tag multiple time by mistake.

2. Go against the Consul API and make everything a `Set<String>`, potentially hiding duplicate tags on service reads call (although there are no such calls currently in the Armeria codebase).

3. Keep the current approach of having a public API that only allows a `Set<String>` of tags to reflect the most common user intent, but internally use a `List<String>` to match the Consul API.

Let me know which one is preferable.

Reply:
Thanks for the kind words. I didn't know the detailed specification. In that case, I lean toward option 3.

Reply:
Thanks for the explanation. 😄 Yeah, I'm fine as it is.